DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+
+        public CryptoHashProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var cryptoHashSetting = settingFactory.CreateAnonymizerSetting<CryptoHashSetting>(settingObject);
+            _cryptoHashFunction = new CryptoHashFunction(cryptoHashSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (item is DicomStringElement)
+            {
+                var encryptedValues = ((DicomStringElement)item).Get<string[]>().Select(GetCryptoHashString);
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptedValues.ToArray());
+            }
+            else if (item is DicomOtherByte)
+            {
+                var valueBytes = ((DicomOtherByte)item).Get<byte[]>();
+                var encryptesBytes = _cryptoHashFunction.ComputeHmacSHA256Hash(valueBytes);
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptesBytes);
+            }
+            else if (item is DicomFragmentSequence)
+            {
+                var enumerator = ((DicomFragmentSequence)item).GetEnumerator();
+
+                var element = item.ValueRepresentation == DicomVR.OW
+                    ? (DicomFragmentSequence)new DicomOtherWordFragment(item.Tag)
+                    : new DicomOtherByteFragment(item.Tag);
+
+                while (enumerator.MoveNext())
+                {
+                    element.Fragments.Add(new MemoryByteBuffer(_cryptoHashFunction.ComputeHmacSHA256Hash(enumerator.Current.Data)));
+                }
+
+                dicomDataset.AddOrUpdate(element);
+            }
+            else
+            {
+                throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationFunction, $"CryptoHash is not supported for {item.ValueRepresentation}");
+            }
+        }
+
+        public bool IsSupportedVR(DicomItem item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            var supportedVR = Enum.GetNames(typeof(CryptoHashSupportedVR)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
>ToHashSet [](http://example.com/codeflow?start=75&length=9)

same, you can generate the hashset in initialization stage and don't need to call it again.

Reply:
Fixed it , Thanks!